src/fhirtypes/ElementDefinition.ts

```diff
@@ -411,9 +409,9 @@ export class ElementDefinition {
    * @throws {InvalidSumOfSliceMinsError} when the sum of mins of the slices exceeds max of sliced element
    */
   private checkSumOfSliceMins(newSlicedElementMax: string, sliceMinIncrease = 0) {
-    const slices = this.structDef.elements.filter(
-      e => e.id !== this.id && e.path === this.path && !e.slicing
-    );
+    const slices = this.parent()
+      .children()
+      .filter(e => e.id !== this.id && e.path === this.path && !e.slicing);
```

Comment:
I think this is just an optimization, right? Functionally the same?

Reply:
Sorry I should have mentioned this actually. This is not functionally the same, there was in fact a bug that I noticed when testing on fsh-mcode. Let's say we had the following four lines.
```
* component[GeneStudied].value[x] only CodeableConcept
* component[GeneStudied].valueCodeableConcept 1..1
* component[VariationCode].value[x] only CodeableConcept
* component[VariationCode].valueCodeableConcept 1..1
```
In the last line, when we try to set the cardinality, we need to check we aren't setting the min too high, so we need to get all slices of `component[VariationCode].value[x]`. But since slices aren't reflected in the path, the previous method would get all elements with path `component.value[x]`, which includes the `component[GeneStudied].value[x]:valueCodeableConcept` slice, which we don't want. Using the new method, we only look at children of `component[VariationCode]`, so we only get the `component[VariationCode].value[x]:valueCodeableConcept` slice that we want.

Reply:
Got it.  I forgot the parent would just be the slice and not the whole array.  It makes sense to me now!